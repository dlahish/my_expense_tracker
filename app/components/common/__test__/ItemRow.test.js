import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import ItemRow from '../ItemRow'

describe('Item Row Test', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ItemRow />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render in edit mode', () => {
    const tree = renderer.create(
      <ItemRow editMode={true} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
