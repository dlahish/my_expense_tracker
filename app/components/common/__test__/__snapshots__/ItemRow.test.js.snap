exports[`Item Row Test render in edit mode 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingLeft": 10,
    }
  }>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingRight": 10,
        }
      }>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "alignSelf": "stretch",
                "borderRadius": 5,
                "flex": 1,
                "height": 44,
                "justifyContent": "center",
                "overflow": "hidden",
              },
              undefined,
            ],
          ]
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 18,
                "margin": 5,
                "textAlign": "center",
              },
              Object {
                "color": "#000",
              },
            ]
          }>
          <View>
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "red",
                    "fontSize": 22,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }>
              ï‡»
            </Text>
          </View>
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#c8c7cc",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "paddingBottom": 8,
            "paddingRight": 15,
            "paddingTop": 8,
          },
        ]
      }
      testID={undefined}>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 17,
                    "fontWeight": "400",
                  },
                  undefined,
                ]
              } />
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 17,
                    "fontWeight": "400",
                  },
                  undefined,
                  undefined,
                ]
              } />
          </View>
        </View>
        <View
          style={
            Object {
              "paddingBottom": 2,
            }
          } />
      </View>
    </View>
  </View>
  <View>
    <View />
  </View>
</View>
`;

exports[`Item Row Test renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingLeft": 10,
    }
  }>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "row",
        },
      ]
    }>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingRight": 10,
        }
      }>
      <View
        style={
          Object {
            "width": 19,
          }
        } />
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderBottomColor": "#c8c7cc",
            "borderBottomWidth": 0.5,
            "flex": 1,
            "paddingBottom": 8,
            "paddingRight": 15,
            "paddingTop": 8,
          },
        ]
      }
      testID={undefined}>
      <View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 17,
                    "fontWeight": "400",
                  },
                  undefined,
                ]
              } />
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#333",
                    "fontSize": 17,
                    "fontWeight": "400",
                  },
                  undefined,
                  undefined,
                ]
              } />
          </View>
        </View>
        <View
          style={
            Object {
              "paddingBottom": 2,
            }
          } />
      </View>
    </View>
  </View>
  <View>
    <View />
  </View>
</View>
`;
